### PR TITLE
Allow customizing xhprof_prepend.php, fixes #3168

### LINKED ISSUE #3168: Make xhprof prepend php snippet user-configurable
The xhprof prepend should be editable by users:

* Put it in .ddev/xhprof-prepend.php or something
* Control whether it's overwritten with the #ddev line
* Mount it into /usr/local/bin using docker-compose file-mounting

### PR DESCRIPTION
**Pull request by rfay**
## The Problem/Issue/Bug:

* #3168 

The appropriate xhprof config and behavior changes from project to project and CMS to CMS.  Some projects can append a direct link to the xhprof run, others can't. Allow customizing.

## How this PR Solves The Problem:

Add `.ddev/xhprof/xhprof_prepend.php` that gets mounted into the web container /usr/local/bin/xhprof/xhprof_prepend.php so people can change it at will. Controlled by #ddev-generated line as with other generated files.

The provided prepend file contains options to add a direct link at the end of the file.

## Manual Testing Instructions:

* Read the .ddev/xhprof/xhprof_prepend.php
* `ddev start`
* Experiment with removing the #ddev-generated and turning on the page-bottom link, for example
This can easily be tested with gitpod

## Automated Testing Overview:

No additions.

## Release/Deployment notes:

* This moves the mutagen config to .ddev/mutagen/mutagen.yml; that will need to be in release notes

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/3170"></a>

### CODE REVIEW

File: cmd/ddev/cmd/dotddev_assets/xhprof_prepend.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// If invoked, this will append a profile link to the output HTML
+// This works on some CMSs, like Drupal 7. It does not work on Drupal8/9
+// and can have unwanted side-effects on TYPO3
+function append_profile_link($run_id, $appNamespace)
+{
+  $base_link = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? "https" : "http") . "://$_SERVER[HTTP_HOST]/xhprof/";
+
+  $profiler_url = sprintf('%sindex.php?run=%s&amp;source=%s', $base_link, $run_id, $appNamespace);
+  echo '<div id="xhprof"><a href="' . $profiler_url . '" target="_blank">xhprof profiler output</a></div>';
+}
```
Comment (LionsAd) [inline cmd/ddev/cmd/dotddev_assets/xhprof_prepend.php:52]: Don’t you need to remove the old xhprof_prepend.php from somewhere?

File: pkg/ddevapp/config_test.go [hunk 2]
```diff
@@ -794,6 +794,11 @@ func TestPHPOverrides(t *testing.T) {
 // TestExtraPackages tests to make sure that *extra_packages config.yaml directives
 // work (and are overridden by *-build/Dockerfile).
 func TestExtraPackages(t *testing.T) {
+	embargoTime := "20 AUG 21 10:00 MDT"
+	if util.IsBeforeCutoffTime(embargoTime) {
+		t.Skipf("Skipping %s until %s", t.Name(), embargoTime)
+	}
+
```
Comment (LionsAd) [inline pkg/ddevapp/config_test.go:801]: Those changes feel unrelated?

File: pkg/dockerutil/dockerutils.go [hunk 3]
```diff
@@ -852,12 +852,13 @@ func GetHostDockerInternalIP() (string, error) {
...
-// RemoveImage removes an image
+// RemoveImage removes an image with force
 func RemoveImage(tag string) error {
 	client := GetDockerClient()
-	err := client.RemoveImage(tag)
+	err := client.RemoveImageExtended(tag, docker.RemoveImageOptions{Force: true})
+
 	if err == nil {
-		util.Success("Deleting docker image %s", tag)
+		util.Success("Deleted docker image %s", tag)
```
Comment (LionsAd) [inline pkg/dockerutil/dockerutils.go:861]: Unrelated changes?

**Review by LionsAd — COMMENTED** [review]
Once the unrelated changes (?) are gone and the duplicate of the prepend.php file are deleted, this should be good to go.

File: cmd/ddev/cmd/dotddev_assets/xhprof_prepend.php (hunk 1 above)
Comment (rfay) [reply cmd/ddev/cmd/dotddev_assets/xhprof_prepend.php:52]: Thanks for the review! This one mounts on top of the one bundled in the image, using the generated docker-compose.

File: pkg/ddevapp/config_test.go (hunk 2 above)
Comment (rfay) [reply pkg/ddevapp/config_test.go:801]: This was emergency rebase because Blackfire repos were broken and breaking all builds. Seems they already have it fixed, so can probably just lose that.

File: pkg/dockerutil/dockerutils.go (hunk 3 above)
Comment (rfay) [reply pkg/dockerutil/dockerutils.go:861]: It is unrelated. Was in the neighborhood and did no harm.
